Code/Framework/AtomCore/AtomCore/Instance/InstanceDatabase.h

```diff
@@ -385,6 +366,57 @@ namespace AZ
             }
         }
 
+        template<typename Type>
+        Data::Asset<Data::AssetData> InstanceDatabase<Type>::LoadAsset(const Data::Asset<AssetData>& asset) const
```

Comment:
Curious why the argument needs to be const when it's not being treated as const in the function body.

Reply:
It doesn't _need_ to be, and I actually think we should change this to be non-const to avoid the copy when possible (it turns out, copying an Asset<T> is annoyingly expensive due to an RTTI type check and a call to update the 'AssetInfo').

I didn't do this at first because we have some places that do something like `RPI::Buffer::FindOrCreate(bufferAssetView.GetBufferAsset());`, where GetBufferAsset returns a const Asset<T>& so it needs a const Asset<T>& version of FindOrCreate. To actually get the benefit out of it, we'd need to go through the higher level functions like Buffer::Create and ShaderResourceGroup::Create so that those functions also have a non-const& version. Since there are a lot of those, not just a single function to update in the InstanceDatabase, I think that warrants a separate change.